packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js

```diff
@@ -13,14 +13,29 @@ import SettingsModal from '../SettingsModal';
 
 const ModalEdit = ({ localeToEdit, onClose, locales }) => {
   const { isEditing, editLocale } = useEditLocale();
+  const [shouldUpdateMenu, setShouldUpdateMenu] = useState(false);
+  const { updateMenu } = useGlobalContext();
   const { formatMessage } = useIntl();
   const isOpened = Boolean(localeToEdit);
 
   const handleSubmit = ({ displayName, isDefault }) => {
     const id = localeToEdit.id;
     const name = displayName || localeToEdit.code;
 
-    return editLocale(id, { name, isDefault }).then(onClose);
+    return editLocale(id, { name, isDefault })
+      .then(() => {
+        console.log('ici');
+        setShouldUpdateMenu(true);
```

Comment:
Why do we need this setSHouldUpdateMenu?

Also, the log 😛

Reply:
Yes, in order to refectch the menu only if we have made some modifications to a locale.